kernel/kernel-api/src/main/java/io/delta/kernel/Table.java

```diff
@@ -34,10 +34,8 @@ public interface Table {
      * @param path        location where the Delta table is present. Path is resolved to fully
      *                    qualified path using the given {@code tableClient}.
      * @return an instance of {@link Table} representing the Delta table at given path
-     * @throws TableNotFoundException when there is no Delta table at the given path.
      */
-    static Table forPath(TableClient tableClient, String path)
-        throws TableNotFoundException {
+    static Table forPath(TableClient tableClient, String path) {
```

Comment:
Can we update these docs about the expectation of the `path`? I.e. it doesn't exist vs does and is empty vs is a delta table etc...

For a new table using Kernel, we are responsible for creating the table directory?